e2e/local-e2e.sh

```diff
@@ -0,0 +1,115 @@
+#!/bin/bash
+
+################################################################################
+#
+# This script is meant to be run from the root of this project
+#
+################################################################################
+
+set -euo pipefail
+
+function run() {
+  minikube_up
+  generate_manifests
+  deploy
+}
+
+# Create local minikube cluster and deploys the dev env for parca and parca agent
+function minikube_up(){
+  echo "Spinning up a parca dev cluster"
+  minikube start -p parca-e2e \
+    --container-runtime=docker \
+    --insecure-registry="localhost:5000"\
+    --driver=kvm2 \
+    --kubernetes-version=v1.22.3 \
+    --cpus=4 \
+    --memory=16gb \
+    --disk-size=20gb \
+    --docker-opt dns=8.8.8.8 \
+    --docker-opt default-ulimit=memlock=9223372036854775807:9223372036854775807
+
+  eval $(minikube -p parca-e2e docker-env)
+}
+
+# Delete minikube instance
+function minikube_down(){
+  echo "Deleting parca-e2e cluster"
+  minikube delete -p parca-e2e
+}
+
+# Configure clusters to run latest commit in Parca agent
+function deploy() {
+  SERVER_LATEST_VERSION=$(curl -s https://api.github.com/repos/parca-dev/parca/releases/latest | grep -oP '"tag_name": "\K(.*)(?=")' | xargs echo -n)
+  echo "Server version: $SERVER_LATEST_VERSION"
+
+  if ! check_ns_parca; then
+    kubectl create namespace parca
+  fi
+
+  kubectl apply -f https://github.com/parca-dev/parca/releases/download/"$SERVER_LATEST_VERSION"/kubernetes-manifest.yaml
+  while ! is_rollout_finished "deployment/parca" ; do
```

Comment:
`kubectl rollout status deployment` should already suppose to wait until deployment actually rollout out. The loop seemed overkill.

Also, there's a `--timeout` flag we should specify a reasonable timeout value to prevent this script to run forever.